snuba/web/rpc/timeseries.py

```diff
@@ -34,19 +36,43 @@
 def _get_aggregate_func(
     request: AggregateBucketRequest,
 ) -> Expression:
-    key_col = attribute_key_to_expression(request.key)
+    key_expr = attribute_key_to_expression(request.key)
+    exists_condition: Expression = literal(True)
+    if request.key.name not in NORMALIZED_COLUMNS:
+        if request.key.type == AttributeKey.TYPE_STRING:
+            exists_condition = f.mapContains(
+                column("attr_str"), literal(request.key.name)
+            )
+        else:
+            exists_condition = f.mapContains(
+                column("attr_num"), literal(request.key.name)
+            )
+    sampling_weight_expr = column("sampling_weight_2")
+    sign_expr = column("sign")
+    sampling_weight_times_sign = f.multiply(sampling_weight_expr, sign_expr)
+
     if request.aggregate == AggregateBucketRequest.FUNCTION_SUM:
-        return f.sum(key_col, alias="sum")
-    if request.aggregate == AggregateBucketRequest.FUNCTION_AVERAGE:
-        return f.avg(key_col, alias="avg")
+        return f.sum(f.multiply(key_expr, sampling_weight_times_sign), alias="sum")
```

Comment:
Does this make all the functions the weighted ones by default using `sampling_weight_2`? Isn't that going to cause some confusion since that column is empty for older data right now